Dannydevs pt 2: Ch. 7 edits (#118)
MystenLabs/move-book: book/src/move-basics/string.md

module defines a `String` type and methods for UTF-8 encoded strings, and the second module,
`std::ascii`, provides an ASCII `String` type and its methods.

> > The Sui execution environment automatically converts bytevector into `String` in transaction inputs.

Reviewer: ```suggestion
> The Sui execution environment automatically converts `bytevector` into `String` in transaction inputs.
```

returns an `Option<String>`, which contains no value if the bytes are not valid UTF-8, and a string
otherwise.

> > Hint: Functions with names starting with try_* typically return an Option. If the operation succeeds, the 

Reviewer: ```suggestion
> Hint: Functions with names starting with `try_*` typically return an Option. If the operation succeeds, the 
```
why remove?

Author: Sorry, might have done it by accident (probably pasted in from chatGPT and it loses the ``). Fixed.

> > result is wrapped in `Some`. If it fails, the function returns `None`. This naming convention, commonly 

Reviewer: This has been written as `some`, Some, and `Some`, same with None. Suggest picking one and using throughout.

Author: Agreed! Did a global find for `some` and `none` and changed them to `Some` and `None`.

4 bytes. Similarly, the `length()` method returns the number of bytes in the string, not the number
of characters.

> However, methods like `sub_string` and `insert` validate character boundaries and will abort if the 

Reviewer: ```suggestion
However, methods like `sub_string` and `insert` validate character boundaries and aborts if the 
```

Author: the subject "methods" is plural, so I'll keep the verb as "abort" -- but I deleted the "will".
